### ISSUE TITLE
OPML import fails when browser supplies incorect MIME

### ISSUE THREAD

**Issue by Brutus5000** [fossar/selfoss#730]
Hi,

I tried to import my feeds from AOL Reader. You can export your feeds into a valid OPML-file.
However, the export file is created with ending .opml
Selfoss only accepts .xml, otherwise the import fails with error message

> There was a problem importing your OPML file:
> Unsupported file type: application/octet-stream

Importing by renaming the file to .xml works. Is this intended behaviour?

**Comment by theomega** [comment]
This depends on the mime-detection on your server. You need to map .opml to the right mime-type `application/xml` in your webserver.

**Comment by Brutus5000** [comment]
Thanks for the info. On hosted environments it's not in the hand of the users.

I stubled across the f3 lib due to another issue. It seems like the f3 lib uses it's own mapping in web.php. Unfortunately it seems to be not in use for this case. I added a line for opml but the issue remains.

It's not a big thing though - I think most user usually import their feeds only once.

```php
	/**
	*	Detect MIME type using file extension
	*	@return string
	*	@param $file string
	**/
	function mime($file) {
		if (preg_match('/\w+$/',$file,$ext)) {
			$map=array(
				'au'=>'audio/basic',
				'avi'=>'video/avi',
				'bmp'=>'image/bmp',
				'bz2'=>'application/x-bzip2',
				'css'=>'text/css',
				'dtd'=>'application/xml-dtd',
				'doc'=>'application/msword',
				'gif'=>'image/gif',
				'gz'=>'application/x-gzip',
				'hqx'=>'application/mac-binhex40',
				'html?'=>'text/html',
				'jar'=>'application/java-archive',
				'jpe?g'=>'image/jpeg',
				'js'=>'application/x-javascript',
				'midi'=>'audio/x-midi',
				'mp3'=>'audio/mpeg',
				'mpe?g'=>'video/mpeg',
				'ogg'=>'audio/vorbis',
				'pdf'=>'application/pdf',
				'png'=>'image/png',
				'ppt'=>'application/vnd.ms-powerpoint',
				'ps'=>'application/postscript',
				'qt'=>'video/quicktime',
				'ram?'=>'audio/x-pn-realaudio',
				'rdf'=>'application/rdf',
				'rtf'=>'application/rtf',
				'sgml?'=>'text/sgml',
				'sit'=>'application/x-stuffit',
				'svg'=>'image/svg+xml',
				'swf'=>'application/x-shockwave-flash',
				'tgz'=>'application/x-tar',
				'tiff'=>'image/tiff',
				'txt'=>'text/plain',
				'wav'=>'audio/wav',
				'xls'=>'application/vnd.ms-excel',
				'xml'=>'application/xml',
				'zip'=>'application/x-zip-compressed'
```

**Comment by jtojnar** [comment]
Selfoss uses the MIME type provided by the web server ([relevant code](https://github.com/SSilence/selfoss/blob/e419e1f6cf3191138a55b32eaf053224d889e396/controllers/Opml.php#L65)) which the server actually gets from the browser. We could try to figure the file type out ourselves but you are right, it probably is not worth the trouble.

**Comment by jtojnar** [comment]
Do we need to check the MIME type? We could just try to process the file.
